src/domain/chains/entities/schemas/__tests__/chain.schema.spec.ts

```diff
@@ -0,0 +1,503 @@
+          {
+            min: 3,
+            max: 9,
+          },
+        )
+        .map((builder) => builder().build());
+
+      const result = GasPriceSchema.safeParse(gasPrice);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should not validate an invalid gas price', () => {
+      const gasPrice = [
+        { invalid: 'gasPriceOracle' },
+        { invalid: 'gasPriceFixed' },
+        { invalid: 'gasPriceFixedEip1559' },
+      ];
+
+      const result = GasPriceSchema.safeParse(gasPrice);
+
+      expect(!result.success && result.error).toStrictEqual(
+        new ZodError([
+          {
+            code: 'invalid_union_discriminator',
+            options: ['oracle', 'fixed', 'fixed1559'],
+            path: [0, 'type'],
+            message:
+              "Invalid discriminator value. Expected 'oracle' | 'fixed' | 'fixed1559'",
+          },
+          {
+            code: 'invalid_union_discriminator',
+            options: ['oracle', 'fixed', 'fixed1559'],
+            path: [1, 'type'],
+            message:
+              "Invalid discriminator value. Expected 'oracle' | 'fixed' | 'fixed1559'",
+          },
+          {
+            code: 'invalid_union_discriminator',
+            options: ['oracle', 'fixed', 'fixed1559'],
+            path: [2, 'type'],
+            message:
+              "Invalid discriminator value. Expected 'oracle' | 'fixed' | 'fixed1559'",
+          },
+        ]),
+      );
+    });
+  });
+
+  describe('ChainSchema', () => {
+    it('should validate a valid chain', () => {
+      const chain = chainBuilder().build();
+
+      const result = ChainSchema.safeParse(chain);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should allow undefined chainLogoUri and ensRegistryAddress and default to null', () => {
+      const chain = chainBuilder().build();
+      // @ts-expect-error - inferred types don't allow optional types
+      delete chain.chainLogoUri;
+      // @ts-expect-error - inferred types don't allow optional types
+      delete chain.ensRegistryAddress;
+
+      const result = ChainSchema.safeParse(chain);
+
+      expect(result.success && result.data.chainLogoUri).toBe(null);
+      expect(result.success && result.data.ensRegistryAddress).toBe(null);
+    });
+
+    it('should checksum the ensRegistryAddress', () => {
+      const nonChecksummedAddress = faker.finance
+        .ethereumAddress()
+        .toLowerCase() as `0x${string}`;
+      const chain = chainBuilder()
+        .with('ensRegistryAddress', nonChecksummedAddress)
+        .build();
+
+      const result = ChainSchema.safeParse(chain);
+
+      expect(result.success && result.data.ensRegistryAddress).toBe(
+        getAddress(nonChecksummedAddress),
+      );
+    });
+
+    it('should not validate an invalid chain', () => {
+      const chain = { invalid: 'chain' };
+
+      const result = ChainSchema.safeParse(chain);
+
+      expect(!result.success && result.error).toStrictEqual(
```

Comment:
nitpick: you can also parameterise this test so that we know which test fails. The `omit` function returns the instance of an object with a specific property omitted (this version is not as straightforward as having a collection of errors listed but the main goal is to isolate each property on a different test).

```typescript
function omit<T>(obj: T, key: keyof T): Partial<T> {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const { [key]: omitted, ...rest } = obj;
  return rest as Partial<T>;
}


// testing required properties only
[
  'chainId',
  'chainName',
  'description',
  'shortName',
  'transactionService',
  'vpcTransactionService',
  'recommendedMasterCopyVersion',
]
  .map((key) => ({
    key,
    input: omit(chainBuilder().build(), key as keyof Chain),
  }))
  .forEach(({ key, input }) => {
    it(`should not validate a chain without ${key}`, () => {
      const result = ChainSchema.safeParse(input);

      expect(!result.success && result.error).toStrictEqual(
        new ZodError([
          {
            code: 'invalid_type',
            expected: 'string',
            received: 'undefined',
            path: [key],
            message: 'Required',
          },
        ]),
      );
    });
  });
```

Reply:
I adjusted the suggestion to not need `omit` by having them in separate `each` tests in d01062e0911503731ad05315755886f3171286e6. Let me know what you think.